pkg/build/builder/dockerutil.go

```diff
@@ -89,6 +120,35 @@ func pushImage(client DockerClient, name string, authConfig docker.AuthConfigura
 	return err
 }
 
+func getDescriptionOfDeniedError(eventWatch watchapi.Interface, ref imageapi.DockerImageReference, err error) error {
+	glog.V(4).Infof("Push has been denied. Waiting for an event with proper error description.")
+	timeout := time.After(time.Second * 2)
+
+waitLoop:
+	for {
```

Comment:
@smarterclayton I don't want to take the `cmd` package dependency here, but I want your `Until` function.  Has that moved  upstream yet? 

@miminar add a TODO to help us find and fix this once we have that in a place we can depend on.